### ISSUE TITLE
Compiling under MacOSX fails

### ISSUE THREAD

**Issue by Matthies** [Matthies/RubiChess#248]
Reported in http://talkchess.com/forum3/viewtopic.php?f=2&t=75016&start=70:
```make COMP=clang
rm -f RubiChess-BMI2 RubiChess-AVX2 RubiChess RubiChess-Legacy *.o
rm -f -rf OPT
rm -f *.clangprof-raw *.profdata
  Compiling RubiChess...
clang++ -std=c++11 -Wall -O3 -flto -pedantic -Wextra -Wshadow  -mssse3 -msse2 -mmmx -mpopcnt *.cpp -lpthread  -D GITVER=\"20201029193215\" -D GITID=\"4cd1205\" -DUSE_SSSE3 -DUSE_SSE2 -DUSE_MMX -DUSE_POPCNT -o RubiChess
board.cpp:2577:31: error: use of undeclared identifier 'aligned_alloc'
    sthread = (searchthread*) allocalign64(size);
                              ^
./RubiChess.h:125:25: note: expanded from macro 'allocalign64'
#define allocalign64(x) aligned_alloc(64, x)
                        ^
1 error generated.
nnue.cpp:429:22: error: use of undeclared identifier 'aligned_alloc'
    bias = (int16_t*)allocalign64(allocsize);
                     ^
./RubiChess.h:125:25: note: expanded from macro 'allocalign64'
#define allocalign64(x) aligned_alloc(64, x)
                        ^
nnue.cpp:431:24: error: use of undeclared identifier 'aligned_alloc'
    weight = (int16_t*)allocalign64(allocsize);
                       ^
./RubiChess.h:125:25: note: expanded from macro 'allocalign64'
#define allocalign64(x) aligned_alloc(64, x)
                        ^
nnue.cpp:467:22: error: use of undeclared identifier 'aligned_alloc'
    bias = (int32_t*)allocalign64(allocsize);
                     ^
./RubiChess.h:125:25: note: expanded from macro 'allocalign64'
#define allocalign64(x) aligned_alloc(64, x)
                        ^
nnue.cpp:469:23: error: use of undeclared identifier 'aligned_alloc'
    weight = (int8_t*)allocalign64(allocsize);
                      ^
./RubiChess.h:125:25: note: expanded from macro 'allocalign64'
#define allocalign64(x) aligned_alloc(64, x)
                        ^
4 errors generated.
transposition.cpp:167:36: error: use of undeclared identifier 'aligned_alloc'
    table = (transpositioncluster*)allocalign64(allocsize);
                                   ^
./RubiChess.h:125:25: note: expanded from macro 'allocalign64'
#define allocalign64(x) aligned_alloc(64, x)
                        ^
transposition.cpp:352:31: error: use of undeclared identifier 'aligned_alloc'
    table = (S_PAWNHASHENTRY*)allocalign64(tablesize);
                              ^
./RubiChess.h:125:25: note: expanded from macro 'allocalign64'
#define allocalign64(x) aligned_alloc(64, x)
                        ^
transposition.cpp:388:33: error: use of undeclared identifier 'aligned_alloc'
    table = (Materialhashentry*)allocalign64(tablesize);
                                ^
./RubiChess.h:125:25: note: expanded from macro 'allocalign64'
#define allocalign64(x) aligned_alloc(64, x)
                        ^
3 errors generated.
make[1]: *** [compile] Error 1
make: *** [default] Error 2
```
And lld is not available for MacOSX

**Comment by Matthies** [comment]
@olithink Don't know if my message via talkches/send email reached you (seemed that the mail was sent to me) so here another try:
Can you replace
--CXXFLAGS=-std=c++11 -Wall -O3 -flto
++CXXFLAGS=-std=c++17 -Wall -O3 -flto
in line 46 of Makefile and see if the aligned_alloc errors are fixed with this?

**Comment by Matthies** [comment]
Via mail: Doesn't work.

**Comment by Matthies** [comment]
https://developer.apple.com/forums/thread/81413
https://stackoverflow.com/questions/44841574/aligned-alloc-not-found-for-clang

**Comment by Matthies** [comment]
Should be fixed with #251
